force-app/main/default/classes/EncryptionRecipes.cls

```diff
@@ -0,0 +1,414 @@
+    }
+
+    /**
+     * @description Encrypts data using AES algorithm, which needs a symmetric key to be shared with the receiver.
+     * In this case the initialization vector is specified by the sender. It needs to be random and 16 bytes (128 bits).
+     * @param dataToEncrypt Blob that contains the data to encrypt
+     * @return Blob
+     * @example
+     * Blob initializationVector = EncryptionRecipes.generateInitializationVector();
+     * Blob dataToEncrypt = Blob.valueOf('Test data');
+     * Blob encryptedData = EncryptionRecipes.encryptRecipe(dataToEncrypt, initializationVector);
+     * System.debug(EncodingUtil.base64Encode(encryptedData));
+     **/
+    @AuraEnabled
+    public static Blob encryptRecipe(
+        Blob dataToEncrypt,
+        Blob initializationVector
+    ) {
+        // Call Crypto.encryptWithManagedIV specifying the selected AES Algorithm
+        Blob encryptedData = Crypto.encrypt(
+            EncryptionRecipes.AESAlgorithm.AES256.name(),
+            AES_KEY,
+            initializationVector,
+            dataToEncrypt
+        );
+        // Combine the encryptedData and initializationVector to send both values to the receiver
+        String blobsAsHex =
+            EncodingUtil.convertToHex(initializationVector) +
+            EncodingUtil.convertToHex(encryptedData);
+        return EncodingUtil.convertFromHex(blobsAsHex);
+    }
+
+    /**
+     * @description Encrypts data using AES algorithm, which needs a symmetric key to be shared with the receiver.
+     * In this case the sender needs to share the initialization vector with the receiver.
+     * @param dataToDecrypt Blob that contains the data to be decrypted
+     * @return Blob
+     * @example
+     * Blob decryptedData = EncryptionRecipes.decryptRecipe(encryptedData);
+     * System.debug(decryptedData.toString());
+     **/
+    @AuraEnabled
+    public static Blob decryptRecipe(Blob dataToDecrypt) {
+        // Convert received blobs to hex
+        String blobsAsHex = EncodingUtil.convertToHex(dataToDecrypt);
+        // Substract first two bytes (initialization vector)
+        String initializationVectorString = blobsAsHex.substring(0, 32);
+        // Substract the rest (encrypted data)
+        String encryptedDataString = blobsAsHex.substring(32);
+        // Convert both hex values to Blob
+        Blob initializationVector = EncodingUtil.convertFromHex(
+            initializationVectorString
+        );
+        Blob encryptedData = EncodingUtil.convertFromHex(encryptedDataString);
+        // Call Crypto.decryptWithManagedIV specifying the selected AES Algorithm
+        return Crypto.decrypt(
+            EncryptionRecipes.AESAlgorithm.AES256.name(),
+            AES_KEY,
+            initializationVector,
+            encryptedData
+        );
+    }
+
+    /**
+     * @description Aux method to generate a random initialization vector.
+     * @return Blob
+     **/
+    public static Blob generateInitializationVector() {
```

Comment:
The IV generation algorithm severely restricts the entropy of the IV since only a subset of the allowed bytes are used. The current algorithm limits the IV to 7 bits per character (vs full 8 bits) and only a subset of printable characters.Please use [java.security.SecureRandom#nextBytes](https://docs.oracle.com/javase/6/docs/api/java/security/SecureRandom.html#nextBytes(byte[]))

Cryptography requires strong random numbers, otherwise the security guarantees of a particular crypto system can be defeated. See the incident regarding Debian SSH keys which didn't have enough entropy https://wiki.debian.org/SSLkeys